api_app/analyzers_manager/observable_analyzers/onionscan.py

```diff
@@ -0,0 +1,57 @@
+"""
+This IntelOwl module adds onionscan utility support to scan tor .onion domains.
+"""
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import json
+import subprocess
+from shutil import which
+
+from api_app.analyzers_manager import classes
+from api_app.exceptions import AnalyzerRunException
+
+
+class OnionScan(classes.ObservableAnalyzer):
+    """
+    Scans domains with onionscan for misconfigurations and leaks
+    """
+
+    onionscan_binary: str = "/opt/deploy/bundled/onionscan"
+    # default target protonmail website
+    target = "protonmailrmez3lotccipshtkleegetolb73fuirgj7r4o4vfu7ozyd.onion"
+
+    def set_params(self, params):
+        """
+        params:
+            - "target" : "target domain for onionscan" (default protonmail)
+        """
+        self.target = params.get("target", self.target)
```

Comment:
this is not how IntelOwl takes inputs. The observable is already available as `self.observable_name`.

Reply:
oh, I'm sorry I simply looked up a few plugins and assumed that. Thanks, you've motivated me to dig deeper into the codebase.